airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -673,6 +674,65 @@ def get_page_size(self) -> Optional[int]:
         return self.page_size
 
 
+@dataclass
+class HubspotCustomObjectsSchemaLoader(SchemaLoader):
+    """
+    Custom schema loader for HubSpot custom object streams.
+
+    This class generates a JSON schema based on the properties defined in the manifest.
+    These properties are injected into the parameters by the HttpComponentsResolver used within the DynamicDeclarativeStream.
+    """
+
+    config: Mapping[str, Any]
+    parameters: InitVar[Mapping[str, Any]]
+
+    def __post_init__(self, parameters: Mapping[str, Any]) -> None:
+        raw_schema_properties: List[Mapping[str, Any]] = parameters.get("schema_properties", {})
+        properties = self._get_properties(raw_schema=raw_schema_properties)
+        self._schema = self._generate_schema(properties)
+
+    def _get_properties(self, raw_schema: List[Mapping[str, Any]]) -> Mapping[str, Any]:
+        return {field["name"]: self._field_to_property_schema(field) for field in raw_schema}
+
+    def _field_to_property_schema(self, field: Mapping[str, Any]) -> Mapping[str, Any]:
+        field_type = field["type"]
+        property_schema = {}
+        if field_type == "enumeration" or field_type == "string":
+            property_schema = {"type": ["null", "string"]}
+        elif field_type == "datetime" or field_type == "date":
```

Comment:
I think `date` is supposed to map to its own distinct schema type based on ["date": {"type": ["null", "string"], "format": "date"}](https://github.com/airbytehq/airbyte-python-cdk/blob/main/airbyte_cdk/sources/declarative/schema/dynamic_schema_loader.py#L25C5-L31)

If I'm understanding this block right in that we're supposed to be mapping Hubspot types back to Airbyte expected types than we should have account for Hubspot's types `datetime` and `date-time` and a separate condition for `date`:
```
elif field_type == "datetime" or field_type == "date-time":
  property_schema = {"type": ["null", "string"], "format": "date-time"}
elif field_type == "date":
  property_schema = {"type": ["null", "string"], "format": "date"}
```